src/juju_doctor/probes.py

```diff
@@ -0,0 +1,62 @@
+"""Helper module to wrap and execute probes."""
+
+import importlib.util
+import inspect
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+
+from rich.logging import RichHandler
+
+from juju_doctor.artifacts import Artifacts
+
+SUPPORTED_PROBE_TYPES = ["status", "bundle", "show_unit"]
+
+logging.basicConfig(level=logging.WARN, handlers=[RichHandler()])
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class Probe:
+    """A probe that can be executed via juju-doctor."""
+
+    name: str
+    uri: str
+    original_path: Path  # path in the source folder
+    path: Path  # path in the temporary folder
+
+    def run(self, artifacts: Artifacts):
+        """Dynamically load a Python script from self.path, making its functions available.
+
+        Execute each Probe function that matches the names: `status`, `bundle`, or `show_unit`.
+
+        We need to import the module dynamically with the 'spec' mechanism because the path
+        of the probe is only known at runtime.
+        """
+        module_name = "probe"
+        # Get the spec (metadata) for Python to be able to import the probe as a module
+        spec = importlib.util.spec_from_file_location(module_name, self.path.resolve())
+        if not spec:
+            raise ValueError(f"Probe not found at its 'path': {self}")
+        # Import the module dynamically
+        module = importlib.util.module_from_spec(spec)
+        if spec.loader:
+            spec.loader.exec_module(module)
+        # Get the functions defined in the probe module
+        functions = inspect.getmembers(module, inspect.isfunction)
```

Comment:
Wdyt about a mapping with pre-filtered keys?
```suggestion
        functions = {name: func for name, func in inspect.getmembers(module, lambda o: inspect.isfunction(o) and o.__name__ in ["status", "bundle", "show_unit"])}
```

Reply:
Done!